tests/providers/cncf/kubernetes/utils/test_pod_manager.py

```diff
@@ -331,6 +332,23 @@ def test_fetch_container_running_follow(
         assert ret.last_log_time == DateTime(2021, 1, 1, tzinfo=Timezone('UTC'))
         assert ret.running is exp_running
 
+    def test_pod_manager_get_client_call_deprecation(self):
+        """Ensure that kube_client.get_kube_client is removed from pod manager in provider 6.0."""
+        try:
+            from airflow.providers.cncf.kubernetes.utils.pod_manager import get_kube_client  # noqa
+        except ImportError:
+            raise Exception(
+                "You must remove this test. It only exists to remind us to remove `get_kube_client`."
+            )
+        from airflow.providers_manager import ProvidersManager
+
+        info = ProvidersManager().providers['apache-airflow-providers-cncf-kubernetes']
+        if semver.VersionInfo.parse(info.version) >= (6, 0):
```

Comment:
Love it. This is cool for deprecating certain features in major version. I guess it will look a bit differently when we want to remove a code that implements a workaround of "min airflow version? 

I thin we should have it as a parameterized tests (and for example pass a lambda ?) to avoid some boierplate ?

Reply:
what kind of lambda are you think of? 

one thing we could do is add `version_info` to ProviderInfo.  that would reduce a little logic here.

Beyond that i'm not seeing a lot of boilerplate to reduce.

i suppose you could add a method like `ProvidersManager.get_provider_version_info(name)` ... and perhaps `ProvidersManager.get_provider_min_airflow_version(name)` for that case.  

also occurred to me that it might be nice to promote `min_airflow_version` to top level on ProviderInfo, though, i am not that familiar with all of the workings so unsure whether that would be problematic.

Reply:
I was thinking about extracting the "from .. import" as lambda in the form of:

```
lambda: __import__('airflow.providers.cncf.kubernetes.utils.pod_manager.get_kube_client')
```

Passed by parameterized. Then there could be one test covering multiple similar cases rather than having to copy  the try/except/raise provider in multiple tests.

Reply:
I see. that makes sense for the use case where the deprecation is removing a function (or, i suppose, anything in globals()) but it might not be that simple if removing a method or a param.  

i don't think you'd need a lambda... you could just parametrize with the import string, and supply it to `import_string`.  

i tried to go with the spirit of your comment in the latest commit. test is _marginally_ more readable.  i reckon should be good enough for now!

could imagine parametrizing but... really the devil is in the details